Using types, return using match fails if f64 is in the list of items? Works if f64 is separate item in the match

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 12.6, 21G115
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 14.0.0 (clang-1400.0.29.201)

V full version: V 0.3.1 f922ed0.71bff21
```
**What did you do?**
`v -g -o vdbg cmd/v && vdbg reperr.v`
```v
type MyT = f32 | f64 | i64 | int | rune | string | u32 | u64
type FnMyT = fn (MyT)

fn (any []MyT) each(fun FnMyT) {
	for x in any {
		fun(x)
	}
}

// single return statements f64 needs to be in separate line???
fn (myt MyT) to_s() string {
	return match myt {
		string { myt }
		rune { rune(myt).str() }
		int, u32, i64, u64, f32, f64 { myt.str() } // FAIL compilation
		// int, u32, i64, u64, f32 { myt.str() }   //  f64 on separate line OK
		// f64 { myt.str() }
	}
}

mut arr := [MyT(1), `😝`, 'we are mamamoo', f32(u64(-1)), f64(u64(-1)), i64(-1), u64(-1)]
arr.each(fn (my MyT) {
	println(my.to_s())
})

```

**What did you expect to see?**

1
😝
we are mamamoo
1.8446744e+19
1.8446744073709552e+19
-1
18446744073709551615

**What did you see instead?**
```
==================
   ^~~~~~~~~~~~~~~
/tmp/v_501/reperr.6232951390683776394.tmp.c:12162:155: error: expected ')'
        string _t1 = ((myt._typ == 20 /* string */)? ((*myt._string)) : (myt._typ == 21 /* rune */)? (rune_str(((rune)((*myt._rune))))) : (int_str((*myt._int))) : (u32_str((*myt._u32))) : (i64_str((*myt._i64))) : (u64_str((*myt._u64))) : (f32_str((*myt._f32))) : (f64_str((*myt._f64))));
                                                                                                                                                                 ^
/tmp/v_501/reperr.6232951390683776394.tmp.c:12162:15: note: to match this '('
        string _t1 = ((myt._typ == 20 /* string */)? ((*myt._string)) : (myt._typ == 21 /* rune */)? (rune_str(((rune)((*myt._rune))))) : (int_str((*myt._int))) : (u32_str((*myt._u32))) : (i64_str((*myt._i64))) : (u64_str((*myt._u64))) : (f32_str((*myt._f32))) : (f64_str((*myt._f64))));
                     ^
1 warning and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.


```

Might be a seperate issue and needs its own entry?  Slightly different problem but related?  
Simpler example but different errors.  Compiles if you split one item per line.

```v
type MyT = int | u32

fn to_s2(myt MyT) string {
	return match myt {
		int, u32 { myt.str() }
	}
}
// ERROR generated
// /tmp/v_501/reserrsimp.7443121169644160983.tmp.c:12083:17: error: expected expression
//         string _t1 = ( : (int_str((*myt._int))) : (u32_str((*myt._u32))));
//                        ^
// 1 warning and 1 error generated.

fn to_s3(myt MyT) {
	println(match myt {
		int, u32 { myt.str() }
	})
}
// ERROR generated
// /tmp/v_501/reserrsimp.2562854170743151420.tmp.c:12083:12: error: expected expression
//         println(( : (int_str((*myt._int))) : (u32_str((*myt._u32)))));
//                   ^
// 1 warning and 1 error generated.
```

--- a/vlib/v/gen/c/match.v
+++ b/vlib/v/gen/c/match.v
@@ -145,7 +145,7 @@ fn (mut g Gen) match_expr_sumtype(node ast.MatchExpr, is_expr bool, cond_var str
 		// iterates through all types in sumtype branches
 		for {
 			g.aggregate_type_idx = sumtype_index
-			is_last := j == node.branches.len - 1
+			is_last := j == node.branches.len - 1 && sumtype_index == branch.exprs.len - 1
 			sym := g.table.sym(node.cond_type)
 			if branch.is_else || (use_ternary && is_last) {
 				if use_ternary {

--- a/vlib/v/tests/match_expr_with_last_branch_aggregate_test.v
+++ b/vlib/v/tests/match_expr_with_last_branch_aggregate_test.v
@@ -1,0 +1,24 @@
+type MyT = f32 | f64 | i64 | int | rune | string | u32 | u64
+type FnMyT = fn (MyT)
+
+fn (any []MyT) each(fun FnMyT) {
+	for x in any {
+		fun(x)
+	}
+}
+
+fn (myt MyT) to_s() string {
+	return match myt {
+		string { myt }
+		rune { rune(myt).str() }
+		int, u32, i64, u64, f32, f64 { myt.str() } // FAIL compilation
+	}
+}
+
+fn test_match_expr_with_last_branch_aggregate() {
+	mut arr := [MyT(1), `😝`, 'we are mamamoo', f32(u64(-1)), f64(u64(-1)), i64(-1), u64(-1)]
+	arr.each(fn (my MyT) {
+		println(my.to_s())
+	})
+	assert true
+}

